Define a virtual MediaSessionManagerInterface interface and have PlatformMediaSessionManager derive from it

eric-carlson opened the pull request:
#### 057e00b5eeccc05da0746b9d8bcaff3211bd4d02
<pre>
Define a virtual MediaSessionManagerInterface interface and have PlatformMediaSessionManager derive from it
<a href="https://bugs.webkit.org/show_bug.cgi?id=291107">https://bugs.webkit.org/show_bug.cgi?id=291107</a>
<a href="https://rdar.apple.com/148623827">rdar://148623827</a>

Reviewed by Andy Estes.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::clientDataBufferingTimerFired):

* Source/WebCore/platform/audio/MediaSessionManagerInterface.h: Added.
(WebCore::MediaSessionManagerInterface::scheduleSessionStatusUpdate):
(WebCore::MediaSessionManagerInterface::hasActiveNowPlayingSession const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingTitle const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingDuration const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingElapsedTime const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingInfoUniqueIdentifier const):
(WebCore::MediaSessionManagerInterface::registeredAsNowPlayingApplication const):
(WebCore::MediaSessionManagerInterface::haveEverRegisteredAsNowPlayingApplication const):
(WebCore::MediaSessionManagerInterface::resetHaveEverRegisteredAsNowPlayingApplicationForTesting):
(WebCore::MediaSessionManagerInterface::prepareToSendUserMediaPermissionRequestForPage):
(WebCore::MediaSessionManagerInterface::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerInterface::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerInterface::configureWirelessTargetMonitoring):
(WebCore::MediaSessionManagerInterface::hasWirelessTargetsAvailable):
(WebCore::MediaSessionManagerInterface::isMonitoringWirelessTargets const):
(WebCore::MediaSessionManagerInterface::addSupportedCommand):
(WebCore::MediaSessionManagerInterface::removeSupportedCommand):
(WebCore::MediaSessionManagerInterface::supportedCommands const):
(WebCore::MediaSessionManagerInterface::resetSessionState):
(WebCore::MediaSessionManagerInterface::updatePresentingApplicationPIDIfNecessary):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::resetRestrictions):
(WebCore::PlatformMediaSessionManager::addRestriction):
(WebCore::PlatformMediaSessionManager::removeRestriction):
(WebCore::PlatformMediaSessionManager::restrictions):
(WebCore::PlatformMediaSessionManager::sessionWillBeginPlayback):
(WebCore::PlatformMediaSessionManager::applicationWillBecomeInactive):
(WebCore::PlatformMediaSessionManager::applicationDidBecomeActive):
(WebCore::PlatformMediaSessionManager::applicationDidEnterBackground):
(WebCore::PlatformMediaSessionManager::applicationWillEnterForeground):
(WebCore::PlatformMediaSessionManager::sessionIsPlayingToWirelessPlaybackTargetChanged):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
(WebCore::PlatformMediaSessionManager::scheduleSessionStatusUpdate): Deleted.
(WebCore::PlatformMediaSessionManager::hasActiveNowPlayingSession const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingTitle const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingDuration const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingElapsedTime const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingInfoUniqueIdentifier const): Deleted.
(WebCore::PlatformMediaSessionManager::registeredAsNowPlayingApplication const): Deleted.
(WebCore::PlatformMediaSessionManager::haveEverRegisteredAsNowPlayingApplication const): Deleted.
(WebCore::PlatformMediaSessionManager::prepareToSendUserMediaPermissionRequestForPage): Deleted.
(WebCore::PlatformMediaSessionManager::willIgnoreSystemInterruptions const): Deleted.
(WebCore::PlatformMediaSessionManager::setWillIgnoreSystemInterruptions): Deleted.
(WebCore::PlatformMediaSessionManager::sessionDidEndRemoteScrubbing): Deleted.
(WebCore::PlatformMediaSessionManager::clientCharacteristicsChanged): Deleted.
(WebCore::PlatformMediaSessionManager::configureWirelessTargetMonitoring): Deleted.
(WebCore::PlatformMediaSessionManager::hasWirelessTargetsAvailable): Deleted.
(WebCore::PlatformMediaSessionManager::isMonitoringWirelessTargets const): Deleted.
(WebCore::PlatformMediaSessionManager::isPlayingToAutomotiveHeadUnit const): Deleted.
(WebCore::PlatformMediaSessionManager::processIsSuspended const): Deleted.
(WebCore::PlatformMediaSessionManager::audioCaptureSourceStateChanged): Deleted.
(WebCore::PlatformMediaSessionManager::audioCaptureSourceCount const): Deleted.
(WebCore::PlatformMediaSessionManager::isInterrupted const): Deleted.
(WebCore::PlatformMediaSessionManager::addSupportedCommand): Deleted.
(WebCore::PlatformMediaSessionManager::removeSupportedCommand): Deleted.
(WebCore::PlatformMediaSessionManager::supportedCommands const): Deleted.
(WebCore::PlatformMediaSessionManager::resetHaveEverRegisteredAsNowPlayingApplicationForTesting): Deleted.
(WebCore::PlatformMediaSessionManager::resetSessionState): Deleted.
(WebCore::PlatformMediaSessionManager::isApplicationInBackground const): Deleted.
(WebCore::PlatformMediaSessionManager::updatePresentingApplicationPIDIfNecessary): Deleted.

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):

* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.h

* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::MediaSessionManageriOS::resetRestrictions):

* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setMediaSessionRestrictions):
(WebCore::Internals::mediaSessionRestrictions const):

* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.cpp:
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationWillEnterForeground):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationDidEnterBackground):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationWillBecomeInactive):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationDidBecomeActive):
(WebKit::RemoteMediaSessionHelperProxy::applicationWillEnterForeground): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationDidEnterBackground): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationWillBecomeInactive): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationDidBecomeActive): Deleted.
* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.h:

Canonical link: <a href="https://commits.webkit.org/293485@main">https://commits.webkit.org/293485@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31b16bc15d28b7b081f649c4d0a7258af0925f08

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18684 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8907 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104152 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49616 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101068 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18955 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27110 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75384 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32511 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102027 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14419 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89431 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55745 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14205 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7409 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48992 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84155 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7485 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106518 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26146 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19046 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84345 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26495 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85629 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83848 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21271 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28511 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6183 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19857 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26073 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31265 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25893 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29213 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27467 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include "MediaUniqueIdentifier.h"
+#include "NowPlayingMetadataObserver.h"
```
Can `NowPlayingMetadataObserver` be forward-declared or is this include truly needed?

aestes commented on Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
```diff
@@ -0,0 +1,156 @@
+namespace WebCore {
+
+class Page;
+class PlatformMediaSession;
```
Seems like this is unnecessary since we include `PlatformMediaSession.h`

aestes commented on Source/WebCore/platform/audio/PlatformMediaSession.h:
```diff
@@ -287,7 +288,7 @@ class PlatformMediaSession
     bool m_isActiveNowPlayingSession { false };
 
-    friend class PlatformMediaSessionManager;
+//    friend class PlatformMediaSessionManager;
```
If this isn't needed can we remove it instead of commenting it out?

eric-carlson committed:
Define a virtual MediaSessionManagerInterface interface and have PlatformMediaSessionManager derive from it
https://bugs.webkit.org/show_bug.cgi?id=291107
rdar://148623827

Reviewed by Andy Estes.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::clientDataBufferingTimerFired):

* Source/WebCore/platform/audio/MediaSessionManagerInterface.h: Added.
(WebCore::MediaSessionManagerInterface::scheduleSessionStatusUpdate):
(WebCore::MediaSessionManagerInterface::hasActiveNowPlayingSession const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingTitle const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingDuration const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingElapsedTime const):
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingInfoUniqueIdentifier const):
(WebCore::MediaSessionManagerInterface::registeredAsNowPlayingApplication const):
(WebCore::MediaSessionManagerInterface::haveEverRegisteredAsNowPlayingApplication const):
(WebCore::MediaSessionManagerInterface::resetHaveEverRegisteredAsNowPlayingApplicationForTesting):
(WebCore::MediaSessionManagerInterface::prepareToSendUserMediaPermissionRequestForPage):
(WebCore::MediaSessionManagerInterface::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerInterface::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerInterface::configureWirelessTargetMonitoring):
(WebCore::MediaSessionManagerInterface::hasWirelessTargetsAvailable):
(WebCore::MediaSessionManagerInterface::isMonitoringWirelessTargets const):
(WebCore::MediaSessionManagerInterface::addSupportedCommand):
(WebCore::MediaSessionManagerInterface::removeSupportedCommand):
(WebCore::MediaSessionManagerInterface::supportedCommands const):
(WebCore::MediaSessionManagerInterface::resetSessionState):
(WebCore::MediaSessionManagerInterface::updatePresentingApplicationPIDIfNecessary):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::resetRestrictions):
(WebCore::PlatformMediaSessionManager::addRestriction):
(WebCore::PlatformMediaSessionManager::removeRestriction):
(WebCore::PlatformMediaSessionManager::restrictions):
(WebCore::PlatformMediaSessionManager::sessionWillBeginPlayback):
(WebCore::PlatformMediaSessionManager::applicationWillBecomeInactive):
(WebCore::PlatformMediaSessionManager::applicationDidBecomeActive):
(WebCore::PlatformMediaSessionManager::applicationDidEnterBackground):
(WebCore::PlatformMediaSessionManager::applicationWillEnterForeground):
(WebCore::PlatformMediaSessionManager::sessionIsPlayingToWirelessPlaybackTargetChanged):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
(WebCore::PlatformMediaSessionManager::scheduleSessionStatusUpdate): Deleted.
(WebCore::PlatformMediaSessionManager::hasActiveNowPlayingSession const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingTitle const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingDuration const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingElapsedTime const): Deleted.
(WebCore::PlatformMediaSessionManager::lastUpdatedNowPlayingInfoUniqueIdentifier const): Deleted.
(WebCore::PlatformMediaSessionManager::registeredAsNowPlayingApplication const): Deleted.
(WebCore::PlatformMediaSessionManager::haveEverRegisteredAsNowPlayingApplication const): Deleted.
(WebCore::PlatformMediaSessionManager::prepareToSendUserMediaPermissionRequestForPage): Deleted.
(WebCore::PlatformMediaSessionManager::willIgnoreSystemInterruptions const): Deleted.
(WebCore::PlatformMediaSessionManager::setWillIgnoreSystemInterruptions): Deleted.
(WebCore::PlatformMediaSessionManager::sessionDidEndRemoteScrubbing): Deleted.
(WebCore::PlatformMediaSessionManager::clientCharacteristicsChanged): Deleted.
(WebCore::PlatformMediaSessionManager::configureWirelessTargetMonitoring): Deleted.
(WebCore::PlatformMediaSessionManager::hasWirelessTargetsAvailable): Deleted.
(WebCore::PlatformMediaSessionManager::isMonitoringWirelessTargets const): Deleted.
(WebCore::PlatformMediaSessionManager::isPlayingToAutomotiveHeadUnit const): Deleted.
(WebCore::PlatformMediaSessionManager::processIsSuspended const): Deleted.
(WebCore::PlatformMediaSessionManager::audioCaptureSourceStateChanged): Deleted.
(WebCore::PlatformMediaSessionManager::audioCaptureSourceCount const): Deleted.
(WebCore::PlatformMediaSessionManager::isInterrupted const): Deleted.
(WebCore::PlatformMediaSessionManager::addSupportedCommand): Deleted.
(WebCore::PlatformMediaSessionManager::removeSupportedCommand): Deleted.
(WebCore::PlatformMediaSessionManager::supportedCommands const): Deleted.
(WebCore::PlatformMediaSessionManager::resetHaveEverRegisteredAsNowPlayingApplicationForTesting): Deleted.
(WebCore::PlatformMediaSessionManager::resetSessionState): Deleted.
(WebCore::PlatformMediaSessionManager::isApplicationInBackground const): Deleted.
(WebCore::PlatformMediaSessionManager::updatePresentingApplicationPIDIfNecessary): Deleted.

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):

* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.h

* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::MediaSessionManageriOS::resetRestrictions):

* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setMediaSessionRestrictions):
(WebCore::Internals::mediaSessionRestrictions const):

* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.cpp:
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationWillEnterForeground):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationDidEnterBackground):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationWillBecomeInactive):
(WebKit::RemoteMediaSessionHelperProxy::uiApplicationDidBecomeActive):
(WebKit::RemoteMediaSessionHelperProxy::applicationWillEnterForeground): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationDidEnterBackground): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationWillBecomeInactive): Deleted.
(WebKit::RemoteMediaSessionHelperProxy::applicationDidBecomeActive): Deleted.
* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.h:

Canonical link: https://commits.webkit.org/293485@main
